docs/upgrade/16_to_17/add_is_published.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/sh
+
+# Elasticsearch details
+ES_HOST="http://localhost:9200"
+INDEX_NAME="opencast_event"
+FIELD_NAME="is_published"
+
+# Elasticsearch authentication
+ES_USER="your_username"
+ES_PASS="your_password"
+
+ES_HEALTH=$(curl -s -u "$ES_USER:$ES_PASS" -X POST "$ES_HOST/_cluster/health" | jq -e '.status')
+
+if [ "$ES_HEALTH" != "green" ] || [ "$ES_HEALTH" != "yellow" ]; then
+  echo "Elasticsearch cluster is not healthy. Please check the cluster status and try again."
+  exit 1
+fi
+
+# Check if the field already exists
+EXISTING_MAPPING=$(curl -s -u "$ES_USER:$ES_PASS" -X GET "$ES_HOST/$INDEX_NAME/_mapping" | jq -r --arg FIELD_NAME "$FIELD_NAME" 'to_entries | .[0].value.mappings.properties[$FIELD_NAME]')
+if [ "$EXISTING_MAPPING" != "null" ]; then
+  echo "Field '$FIELD_NAME' already exists in index '$INDEX_NAME'. No changes needed."
+else
+  # Update index mapping to add the new field
+  echo "Updating index mapping to add '$FIELD_NAME' field..."
+  curl -u "$ES_USER:$ES_PASS" -X PUT "$ES_HOST/$INDEX_NAME/_mapping" -H "Content-Type: application/json" -d '{
+    "properties": {
+      "'"$FIELD_NAME"'": {
+        "type": "boolean"
+      }
+    }
+  }'
+fi
+
+SCRIPT="
+boolean hasExternalPublication = false;
+if (ctx._source.containsKey('publication')) {
+  for (publication in ctx._source.publication) {
+    if (publication.channel != 'internal') {
+      hasExternalPublication = true;
+      break;
+    }
+  }
+}
+ctx._source.is_published = hasExternalPublication;
+"
+
+# Update documents: Set 'is_published' based on publication channel
+echo "Updating documents based on publication channel..."
+curl -u "$ES_USER:$ES_PASS" -X POST "$ES_HOST/$INDEX_NAME/_update_by_query" -H "Content-Type: application/json" -d '{
+  "script": {
+    "source": "'"$(echo "$SCRIPT" | tr '\n' ' ')"'",
+    "lang": "painless"
+  }
+}'
+
+if [ $? != 0 ]; then
```

Comment:
`curl` needs `-f` to fail for HTTP response codes >= 400. I suggest adding `set -e` for the whole script, but since you handle the exit code check here explicitly, I suggest disabling exit on error for the previous command.

Reply:
I refactored the script a little bit and added `--fail` and `--show-error`